store/map/datasets.js

```diff
@@ -11,31 +11,41 @@ const emptyLocationsObject = () => ({})
 const getOrEmpty = empty => when(identity, identity, empty)
 const getOrEmptyVector = getOrEmpty(emptyLocationsObject)
 const getOrEmptyPointData = getOrEmpty(emptyObject)
-const getOrEmptySpatial = getOrEmpty(emptyObject)
+const getOrEmptyRaster = getOrEmpty(emptyObject)
 const getOrEmptyMetadata = getOrEmpty(emptyObject)
 const getPointData = pipe([get('pointData'), getOrEmptyPointData])
 const getVectorData = pipe([get('vector'), getOrEmptyVector])
-const getSpatialData = pipe([get('spatial'), getOrEmptySpatial])
+const getRasterData = pipe([get('raster'), getOrEmptyRaster])
 const getMetadata = pipe([get('metadata'), getOrEmptyMetadata])
 
 export const state = () => ({})
 
 export const mutations = {
   addDatasetVector(state, data) {
-    const id = data.id
+    const id = get('id', data)
+    if (!id) return
     if (!state[id]) Vue.set(state, id, {})
     const vectorData = getVectorData(state[id])
     Vue.set(
       state[id],
       'vector',
-      merge(vectorData, { mapboxLayer: data.mapboxLayer }),
+      // TODO: make generic by looping over vectorLayer
+      merge(vectorData, { mapboxLayer: get('vectorLayer.mapboxLayer', data) }),
     )
   },
-  addDatasetSpatial(state, data) {
-    const id = data.id
+  addDatasetRaster(state, data) {
+    const id = get('id', data)
+    // If id already has a rasterLayer or url is not correct return
+    if (!id || data.rasterLayer.url === null) return
```

Comment:
When would this occur?  Separate configuration check?

Reply:
check inside index.js (where datasets are read out)